robot_sort/robot_sort.py

```diff
@@ -96,8 +96,18 @@ def sort(self):
         """
         Sort the robot's list.
         """
-        # Fill this out
-        pass
+        for step in range(len(l)):
+            key = l[step]
+
+            j = step - 1
+
+            #comparing the key with each element on the left until a smaller element is found
+            while j >= 0 and key < l[j]:
+                l[j + 1] = l[j]
+                j = j - 1
+
+                #placing the key after the element
+                l[j + 1] = key
```

Comment:
Check the `README.md` one more time. There are specific restrictions on what you can and can't do inside this function.

There are many simple ways to sort a list. But the question is, can you sort the list with the given restrictions?

Reply:
Okay I forgot about the variable rule. I got a sorting algorithm in mind.